go/multiorch/recovery/analysis/recovery.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis
+
+import (
+	"context"
+	"time"
+)
+
+// RecoveryAction is a function that fixes a problem.
+type RecoveryAction interface {
+	// Execute performs the recovery.
+	Execute(ctx context.Context, problem Problem) error
+
+	// Metadata returns info about this recovery.
+	Metadata() RecoveryMetadata
+
+	// RequiresLock indicates if tablegroup lock is needed.
+	RequiresLock() bool
+
```

Comment:
Why would we need a lock at the TableGroup level?

Reply:
Good catch. This should be shard level. I will fix